source/src/core/graph.cpp

```diff
@@ -106,24 +106,32 @@ AudioGraph::~AudioGraph()
 
 void AudioGraph::wait(float time)
 {
-    struct timeval tv;
-    gettimeofday(&tv, NULL);
-    double t0 = tv.tv_sec + tv.tv_usec / 1000000.0;
-
-    while (true)
+    // TODO Tim: I've tried to keep the original logic here, but most likely std::this_thread::sleep_for(std::chrono::duration<float>(time)) would make more sense
```

Comment:
Two comments:
(1) `sleep_for` looks like a much more elegant solution here so feel free to swap it in!
(2) If `time` is equal to zero, this function should wait forever, so there will need to be some logic added for that. (This is the paradigm used in most of the examples: create your AudioGraph, set the synthesis going, and then call `graph.wait()` (which has a default value of 0) to spin forever. Tiny bit of syntax that saves the user having to make their own while/sleep cycle.)